documentation/modules/exploit/linux/local/tomcat_ubuntu_log_init_priv_esc.md

```diff
@@ -0,0 +1,128 @@
+## Vulnerable Application
+
+Tomcat (6, 7, 8) packages provided by default repositories on Debian-based
+distributions (including Debian, Ubuntu etc.) provide a vulnerable
+tomcat init script that allows local attackers who have already gained access
+to the tomcat account (for example, by exploiting an RCE vulnerability
+in a java web application hosted on Tomcat, uploading a webshell etc.) to
+escalate their privileges from tomcat user to root and fully compromise the
+target system.
+
+Tested against Tomcat 8.0.32-1ubuntu1.1 on Ubuntu 16.04
+
+### Install
+
+This will install Tomcat 8 (8.0.32-1ubuntu1.1) on Ubuntu 16.04 using the build from
+[launchpad](https://launchpad.net/~ubuntu-security/+archive/ubuntu/ppa/+build/10427282)
+
+We also change the `tomcat8` user's shell to `/bin/bash` to make setting up the priv-esc
+easier.
+
+```
+sudo apt-get install libecj-java build-essential
+wget https://launchpad.net/~ubuntu-security/+archive/ubuntu/ppa/+build/10427282/+files/tomcat8_8.0.32-1ubuntu1.1_all.deb
+wget https://launchpad.net/~ubuntu-security/+archive/ubuntu/ppa/+build/10427282/+files/libtomcat8-java_8.0.32-1ubuntu1.1_all.deb
+wget https://launchpad.net/~ubuntu-security/+archive/ubuntu/ppa/+build/10427282/+files/tomcat8-common_8.0.32-1ubuntu1.1_all.deb
+sudo dpkg -i *tomcat*.deb
+sudo sed -i 's|/bin/false|/bin/bash|g' /etc/passwd
```

Comment:
Should this be restricted to just the tomcat8 account?

Reply:
yes, but this also assumes you made a box for testing this exploit specifically, so there are no real implications with being too broad.

Reply:
Not a blocker; I think we can remove this line if we just update the `su` line below to specify the shell to use explicitly

```
sudo su --shell /bin/bash tomcat8
```